### ISSUE TITLE
colorbar resizes in animation

### ISSUE THREAD

**Issue by sakra1** [matplotlib/matplotlib#12550]
### Bug report
When animating a 2D plot with a colorbar, updating the color limits in each iteration changes the size of the colorbar. This issue did not exist with matplotlib 2.2.3.

```Python
import numpy as np
import matplotlib.pyplot as plt
from scipy.ndimage import laplace


def main():
    time = np.arange(0, 1, 0.01)
    c = np.zeros(shape=(11, 11))
    c[5, 5] = 1

    fig = plt.figure()
    ax = fig.gca()
    im = ax.imshow(c)
    fig.colorbar(im)

    for t in time:
        c += laplace(c) / 10
        im.set_data(c)
        im.set_clim(0, np.max(c))
        plt.pause(0.001)
    plt.show()


if __name__ == "__main__":
    main()
```

**Actual outcome**
![outcome_matplotlib300](https://user-images.githubusercontent.com/15139452/47114424-0bb47480-d25c-11e8-8b6d-3e0bdf9e0b2c.jpg)

**Expected outcome**
![outcome_matplotlib223](https://user-images.githubusercontent.com/15139452/47114445-1a9b2700-d25c-11e8-8059-de802966cab6.jpg)

**Matplotlib version**
  * Operating system: windows 7
  * Matplotlib version: 3.0.0
  * Matplotlib backend (`print(matplotlib.get_backend())`): TkAgg
  * Python version: 3.7

Python was installed using the latest Anaconda distribution.

**Comment by ImportanceOfBeingErnest** [comment]
This bug came in in ENH: adjustable colorbar ticks #9903.

**Comment by ImportanceOfBeingErnest** [comment]
@sakra1 Your bug report is fine, thanks for that. To explain a bit on the structure here:

The bug appears first in the development version after #9903 has been merged on 22 March. (This can be found out via `git bisect`) This commit has been tagged for the 3.0 branch, such that it is not part of the (later) 2.2.3 release. The first release with this bug in it is hence the 3.0.0rc1.

**Comment by jklymak** [comment]
Yes, sorry this broke.  I believe it is fixed in #12159 and on master,  and will be part of 3.0.1 release in early Nov.
